common/sensor/dev/pex89000.c

```diff
@@ -0,0 +1,353 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <string.h>
+#include "sensor.h"
+#include "hal_i2c.h"
+
+#define BRCM_I2C5_CMD_READ 0b100
+#define BRCM_I2C5_CMD_WRITE 0b011
+
+#define BRCM_I2C5_ACCESS_MODE_FULL 2
+#define BRCM_I2C5_ACCESS_MODE_LOWER18 0
+
+#define BRCM_CHIME_AXI_CSR_ADDR 0x001F0100
+#define BRCM_CHIME_AXI_CSR_DATA 0x001F0104
+#define BRCM_CHIME_AXI_CSR_CTL 0x001F0108
+
+#define BRCM_TEMP_SNR0_CTL_REG1 0xFFE78504
+#define BRCM_TEMP_SNR0_STAT_REG0 0xFFE78538
+
+#define BRCM_TEMP_SNR0_CTL_REG1_RESET 0x000653E8
+
+#define BRCM_FULL_ADDR_BIT 0x007C0000 // bits[22:18]
+
+#define TEMP 0xFE // TBD: sensor offset
+
+typedef struct {
+	uint8_t oft9_2bit : 8;
+	uint8_t oft11_10bit : 2;
+	uint8_t be : 4;
+	uint8_t reserve3 : 1;
+	uint8_t oft12bit : 1;
+	uint8_t oft17_13bit : 5;
+	uint8_t access_mode : 2;
+	uint8_t reserve2 : 1;
+	uint8_t cmd : 3;
+	uint8_t reserve1 : 5;
+} __packed __aligned(4) HW_I2C_Cmd;
+
+static void swap32(uint32_t *d)
+{
+	if (!d)
+		return;
+
+	uint32_t t = ((*d >> 24) & 0xFF) | ((*d >> 8) & 0xFF00) | ((*d << 8) & 0xFF0000) |
+		     ((*d << 24) & 0xFF000000);
+
+	*d = t;
+}
+
+/*
+ * be: byte enables
+ * oft: Atlas register address
+ * cmd: read or write command
+ * access_mode: access mode
+ * data: DWORD data
+ * buf: encoded byte array to send to pesw
+ */
+
+static void pex89000_i2c_encode(uint32_t oft, uint8_t be, uint8_t cmd, uint8_t access_mode,
+				HW_I2C_Cmd *buf)
+{
+	buf->reserve1 = 0;
+	buf->cmd = cmd;
+	buf->reserve2 = 0;
+	buf->access_mode = access_mode;
+	buf->oft17_13bit = (oft >> 13) & 0x1F;
+	buf->oft12bit = (oft >> 12) & 0x1;
+	buf->reserve3 = 0;
+	buf->be = be;
+	buf->oft11_10bit = (oft >> 10) & 0x3;
+	buf->oft9_2bit = (oft >> 2) & 0xFF;
+}
+
+static uint8_t set_full_addr(uint8_t bus, uint8_t addr, uint32_t oft)
+{
+	/* I2C write command to set address bits[22:18] in 0x2CC[4:0] */
+
+	uint8_t sub_oft = (uint8_t)(oft & BRCM_FULL_ADDR_BIT) >> 18; // sub_oft: bit[22:18]
+
+	HW_I2C_Cmd cmd;
+	pex89000_i2c_encode(0x2CC, 0xF, BRCM_I2C5_CMD_WRITE, BRCM_I2C5_ACCESS_MODE_LOWER18, &cmd);
+
+	uint8_t data[8] = { 0 };
+	memcpy(&data[0], &cmd, sizeof(cmd));
+	memcpy(&data[7], &sub_oft, sizeof(sub_oft));
+
+	uint8_t retry = 5;
+	I2C_MSG msg;
+
+	msg.bus = bus;
+	msg.slave_addr = addr;
+	msg.tx_len = sizeof(data);
+	memcpy(&msg.data[0], data, sizeof(data));
+
+	if (i2c_master_write(&msg, retry)) {
+		/* write fail */
+		printf("set AXI register to full mode failed!\n");
+		return 0;
+	}
+
+	return 1;
```

Comment:
Generally, a return code of '0' is returned if success and anything else is failure.
Can we swap these to follow normal conventions?